### PR TITLE
Add ES6 modules to JS + build workflow

### PR DESCRIPTION
**Pull request by drublic**
This commit adds support for ES6 modules to the Javascript workflow. It
removes Require.js and adds Browserify and Babel where necessary.

This commit also removes JS unit test coverage tracking which might get
implemented later.

### CODE REVIEW

File: grunt/config.js [hunk 1]
```diff
@@ -14,15 +14,12 @@ module.exports = {
...
-
-  requirejs: '../../node_modules/requirejs/require',
-
   // All files that should be checked with JSHint
   jsHintFiles: [
     'Gruntfile.js',
     'src/**/*.js',
-    'test/*.js',
-    'test/specs/**/*.js'
+    '!src/dist/**/*.js',
```
Comment (anselmh) [inline grunt/config.js]: TBH this doesn’t make sense to me at all. a dist folder in the src should be avoided at all cost IMO. But yes, I’m aware this is another issue not related to this PR.

File: grunt/config.js [hunk 2]
```diff
@@ -31,12 +28,20 @@ module.exports = {
...
+    dev: {
+      entry: 'src/js/main.js',
+      dest: 'src/dist/index.js'
+    },
+    build: {
+      entry: 'src/js/main.js',
+      dest: 'dist/<%= pkg.version %>/main.min.js',
+    },
+    test: {
+      entry: 'test/test-main.js',
```
Comment (anselmh) [inline grunt/config.js]: Should we rename this to main.js as well? Would work in my opinion as it is in the test folder.

File: src/js/main.js [hunk 3]
```diff
@@ -4,12 +4,10 @@
  * @author Author Name
  * @date 2013-01-01
```
Comment (anselmh) [inline src/js/main.js]: This *could* be updated at some point :D

**Comment by anselmh** [comment]
Besides minor things added as inline comments this looks and works fine for me.

File: grunt/config.js (hunk 2 above)
Comment (drublic) [inline grunt/config.js]: Done

**Comment by drublic** [comment]
Thanks for checking this.

File: src/js/main.js (hunk 3 above)
Comment (drublic) [inline src/js/main.js]: Done on branch v2.0

**Pull request merged**